src/runtime_src/core/pcie/tools/xbutil/CMakeLists.txt

```diff
@@ -24,5 +24,7 @@ target_link_libraries(xbutil
   ${CURSES_LIBRARIES}
   )
 
+set_property(TARGET xbutil PROPERTY ENABLE_EXPORTS 1)
+
```

Comment:
xbutil should not export symbols.  I understand the problem, but we need to find a way to not load xdp when xbutil is running.   The proposed fix is just a work-around, isn't it?

Reply:
This export is only for the loadable modules within the executable. It helps the libraries which are dynamically loaded from within the executable. Do you see a potential issue with this?

Also, xbutil query is not loading full "xdp" library. It is loading "xdp_hal_plugin" library as it has been specifically enabled in the xrt.ini with xrt_profile=true under Debug configuration. In addition to this, currently loading of xdp_hal_plugin is actually under a compilation macro "ENABLE_HAL_PROFILING" to be defined in pcie/linux/shim.cpp. The plugin library is not loaded otherwise. As the user has specifically enabled this, we don't have a way to block it except some hardcoded executable name checks.

Reply:
Yes, I know the export thing.  I am not comfortable with this change.  dynamically loaded modules don't necessarily want to resolve symbols against the executable.  Imagine 
```
a.so -> { b.so } // a.so linked with b.so
x.exe  // exporting arbitrary symbols one of happens to be same as one in b.so
```
If x.exe dlopens a.so, will a.so resolve symbols against b.so or x.exe? 
As an example, consider b.so to be a boost library and a.so using a  different version  of boost compared to x.exe.  This could lead to segmentation fault.

I understand what you are doing, and it is safe in our context, but it is just not a good solution, and I really would like to avoid xbutil exporting symbols.

As we have talked about in the past, I don't think it makes sense to hal profile xbutil.  But assume it does and that it can't be prevented why can't xdp_hal resolve  the missing symbols by simply statically with core_static as xrt_core?

Reply:
Thanks for explaining in details.
Linking with xrt_core_static does not help. However, dynamically linking xrt_core with xdp_hal_plugin helps here.
In XDP, we had worked towards removing direct dependency and linking with xrt_core for "xdp_core", "xdp", "xdp_hal_plugin" libraries. We were only linking with xrt_coreutil.
But dynamically linking with xrt_core seems to be the only acceptable solution here.
Another possible solution would be to add the required APIs to "xrt_core::system" class. But it  may not be justifiable for this corner case.